pyomo/contrib/solver/solvers/gurobi_direct.py

```diff
@@ -374,7 +376,12 @@ def solve(self, model, **kwds) -> Results:
             timer,
             config,
             GurobiDirectSolutionLoader(
-                gurobi_model, A, x, repn.rows, repn.columns, repn.objectives
+                gurobi_model,
+                A,
+                x.tolist(),
```

Comment:
Why convert x to a list?

Reply:
Because of the thing you commented on above, actually. I'm making this look the same as Gurobi MINLP so that they can use an identical SolutionLoader. Gurobi MINLP doesn't have the luxury of declaring the variables as a vector (I don't think? Although if I can dynamically add to one, that's another option, maybe?), so they're a list at this point right now.